tidb-binlog/pump_client/client.go

```diff
@@ -420,9 +421,16 @@ func (c *PumpsClient) Close() {
 	Logger.Infof("[pumps client] is closed")
 }
 
-func isCriticalError(err error) bool {
-	// TODO: add some critical error.
-	return false
+func isRetryableError(err error) bool {
+	// ResourceExhausted is a error code in grpc.
+	// ResourceExhausted indicates some resource has been exhausted, perhaps
+	// a per-user quota, or perhaps the entire file system is out of space.
+	// https://github.com/grpc/grpc-go/blob/9cc4fdbde2304827ffdbc7896f49db40c5536600/codes/codes.go#L76
+	if strings.Contains(err.Error(), "ResourceExhausted") {
```

Comment:
why this is not retryable

Reply:
almost all the errer with ResourceExhausted is `received message larger than max`, it is no need to retry